truss/util/path.py

```diff
@@ -75,3 +99,65 @@ def build_truss_shadow_target_directory(stub: str, truss_path: Path) -> Path:
     target_directory_path = Path(Path.home(), ".truss", "models", f"{stub}-{suffix}")
     target_directory_path.mkdir(parents=True, exist_ok=True)
     return target_directory_path
+
+
+def load_trussignore_patterns(truss_ignore_file: Path):
+    """Load patterns from a .truss_ignore file"""
+    patterns = []
+    with truss_ignore_file.open("r") as f:
+        for line in f:
+            line = line.strip()
+            if line and not line.startswith("#"):
+                patterns.append(line)
+    return patterns
+
+
+def is_ignored(path: Path, patterns: List[str]) -> bool:
+    """Check if a given path or any of its parts matches any pattern in .truss_ignore"""
+    while path:
+        for pattern in patterns:
+            if path.is_dir() and pattern.endswith("/"):
+                pattern = pattern.rstrip("/")
+                if fnmatch.fnmatch(path.name, pattern):
+                    return True
+            elif fnmatch.fnmatch(path.name, pattern):
+                return True
+        path = path.parent if path.parent != path else None  # type: ignore
```

Comment:
Can you explain this part. Would this keep going tup until it reaches `/'? Is that what we want?

Reply:
Will be good to make the doc comment more descriptive about the protocol and what each parameter means.